### PR TITLE
Allows rotating statues with alt click

### PR DESCRIPTION
**Pull request by Mickyan**
They're a pain to place properly without this

:cl:
tweak: Statues can be rotated by alt clicking, if unanchored
/:cl:

### CODE REVIEW

File: code/game/objects/structures/statues.dm [hunk 1]
```diff
@@ -18,6 +18,22 @@
 	. = ..()
 	AddComponent(art_type, impressiveness)
 	INVOKE_ASYNC(src, /datum.proc/_AddComponent, list(/datum/component/beauty, impressiveness *  75))
+	AddComponent(/datum/component/simple_rotation,ROTATION_ALTCLICK | ROTATION_CLOCKWISE, CALLBACK(src, .proc/can_user_rotate),CALLBACK(src, .proc/can_be_rotated),null)
+
+/obj/structure/statue/proc/can_be_rotated(mob/user)
+	if(!anchored)
```
Comment (Qustinnus) [inline code/game/objects/structures/statues.dm:24]: give some feedback to the player why they cant rotate if this fails

File: code/game/objects/structures/statues.dm [hunk 2]
```diff
@@ -18,6 +18,22 @@
...
+		return TRUE
+
+/obj/structure/statue/proc/can_user_rotate(mob/user)
+	var/mob/living/L = user
+	if(istype(L))
+		if(!user.canUseTopic(src, BE_CLOSE, ismonkey(user)))
+			return FALSE
+		else
+			return TRUE
+	else if(isobserver(user) && CONFIG_GET(flag/ghost_interaction))
```
Comment (Qustinnus) [inline code/game/objects/structures/statues.dm:34]: why should ghosts be able to rotate statues

Comment (Mickyan) [reply code/game/objects/structures/statues.dm:34]: because I copypasted this from chairs :)

**Comment by Winterous** [comment]
A note: You can presently change the rotation of a statue by dragging it into a solid object, like yourself, to make it face that object. This is definitely a good addition though.

File: code/game/objects/structures/statues.dm [hunk 3]
```diff
@@ -18,6 +18,21 @@
 	. = ..()
 	AddComponent(art_type, impressiveness)
 	INVOKE_ASYNC(src, /datum.proc/_AddComponent, list(/datum/component/beauty, impressiveness *  75))
+	AddComponent(/datum/component/simple_rotation,ROTATION_ALTCLICK | ROTATION_CLOCKWISE, CALLBACK(src, .proc/can_user_rotate),CALLBACK(src, .proc/can_be_rotated),null)
```
Comment (Jared-Fogle) [inline code/game/objects/structures/statues.dm:21]:
```suggestion
	AddComponent(/datum/component/simple_rotation, ROTATION_ALTCLICK | ROTATION_CLOCKWISE, CALLBACK(src, .proc/can_user_rotate), CALLBACK(src, .proc/can_be_rotated), null)
```

File: code/game/objects/structures/statues.dm [hunk 4]
```diff
@@ -18,6 +18,21 @@
...
+	to_chat(user, "<span class='warning'>It's bolted to the floor, you'll need to unwrench it first.</span>")
+
+/obj/structure/statue/proc/can_user_rotate(mob/user)
+	var/mob/living/L = user
+	if(istype(L))
+		if(!user.canUseTopic(src, BE_CLOSE, ismonkey(user)))
+			return FALSE
+		else
+			return TRUE
+	return FALSE
```
Comment (Jared-Fogle) [inline code/game/objects/structures/statues.dm:35]:
```suggestion
	return isliving(user) && user.canUseTopic(src, BE_CLOSE, no_dexterity = ismonkey(user))
```

**Pull request merged**